### PR TITLE
chore: update content type, input body/headers/queryitem

### PR DESCRIPTION
**Pull request by wooj2**
Promoting various middlewares to fully fledged classes :)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/ContentMD5Middleware.kt [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
     fun render(
         op: OperationShape,
         writer: SwiftWriter,
         outputShapeName: String,
         operationStackName: String
     ) {
-        val outputErrorName = "${op.capitalizedName()}OutputError"
-        writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N<$outputShapeName, $outputErrorName>())", ClientRuntimeTypes.Middleware.ContentMD5Middleware)
+        if (op.hasTrait<HttpChecksumRequiredTrait>()) {
+            val outputErrorName = "${op.capitalizedName()}OutputError"
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/ContentMD5Middleware.kt:27]: can we use the static function `getOperationErrorShapeName` inside `ServiceGenerator` to get the name of an output error to keep consistency?

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/ContentTypeMiddleware.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        writer: SwiftWriter,
+        serviceShape: ServiceShape,
+        op: OperationShape,
+        operationStackName: String
+    ) {
+        val inputShape = ctx.model.expectShape(op.input.get())
+        val inputShapeName = ctx.symbolProvider.toSymbol(inputShape).name
+        val outputShape = ctx.model.expectShape(op.output.get())
+        val outputShapeName = ctx.symbolProvider.toSymbol(outputShape).name
+        val outputErrorName = "${op.capitalizedName()}OutputError"
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/ContentTypeMiddleware.kt:31]: same thing here, can we use the static functions inside `ServiceGenerator` to eliminate duplicated code here?

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    override fun render(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+        serviceShape: ServiceShape,
+        op: OperationShape,
+        operationStackName: String
+    ) {
+        val inputShape = ctx.model.expectShape(op.input.get())
+        val inputShapeName = ctx.symbolProvider.toSymbol(inputShape).name
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt:28]: use static functions here please

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt [hunk 4]
```diff
@@ -143,17 +139,6 @@ class MiddlewareExecutionGenerator(
...
-        writer.write("$operationStackName.serializeStep.intercept(position: .before, middleware: ${inputShapeName}QueryItemMiddleware())")
-        writer.write("$operationStackName.serializeStep.intercept(position: .after, middleware: ContentTypeMiddleware<$inputShapeName, $outputShapeName, $outputErrorName>(contentType: \"${defaultContentType}\"))")
-        val hasHttpBody = inputShape.members().filter { it.isInHttpBody() }.count() > 0
-        if (hasHttpBody) {
-            writer.write("$operationStackName.serializeStep.intercept(position: .before, middleware: ${inputShapeName}BodyMiddleware())")
-        }
-        if (op.hasTrait<HttpChecksumRequiredTrait>()) {
-            ContentMD5Middleware().render(op, writer, outputShapeName, operationStackName)
-        }
-
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt:156]: <3

**Review by kneekey23 — APPROVED** [review]
LGTM, love this change, only request is to use our static functions for getting input/output/error names to eliminate duplicated code.

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt (hunk 3 above)
Comment (wooj2) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt:28]: done

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputHeadersMiddleware.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    override fun render(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+        serviceShape: ServiceShape,
+        op: OperationShape,
+        operationStackName: String
+    ) {
+        val inputShape = ctx.model.expectShape(op.input.get())
+        val inputShapeName = ctx.symbolProvider.toSymbol(inputShape).name
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputHeadersMiddleware.kt:27]: update to use ServiceGeneration function please

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputQueryItemMiddleware.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    override fun render(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+        serviceShape: ServiceShape,
+        op: OperationShape,
+        operationStackName: String
+    ) {
+        val inputShape = ctx.model.expectShape(op.input.get())
+        val inputShapeName = ctx.symbolProvider.toSymbol(inputShape).name
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputQueryItemMiddleware.kt:27]: same here

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputHeadersMiddleware.kt (hunk 5 above)
Comment (wooj2) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputHeadersMiddleware.kt:27]: done.

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputQueryItemMiddleware.kt (hunk 6 above)
Comment (wooj2) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputQueryItemMiddleware.kt:27]: done.

**Pull request merged**